pkg/services/preference/prefimpl/pref.go

```diff
@@ -174,6 +178,13 @@ func (s *Service) Patch(ctx context.Context, cmd *pref.PatchPreferenceCommand) e
 		preference.JSONData.Language = *cmd.Language
 	}
 
+	if cmd.Locale != nil {
```

Comment:
qq: why separate the two conditional statements? This applies to multiple files here.

Reply:
I just followed the pattern of `language` tbh